segmentstore/server/src/main/java/io/pravega/segmentstore/server/tables/EntrySerializer.java

```diff
@@ -0,0 +1,221 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.server.tables;
+
+import com.google.common.base.Preconditions;
+import io.pravega.common.io.SerializationException;
+import io.pravega.common.util.ArrayView;
+import io.pravega.common.util.BitConverter;
+import io.pravega.segmentstore.contracts.tables.TableEntry;
+import io.pravega.segmentstore.contracts.tables.TableKey;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.val;
+
+/**
+ * Serializes {@link TableEntry} instances.
+ */
+class EntrySerializer {
+    static final int HEADER_LENGTH = 1 + Integer.BYTES * 2; // Version, Key Length, Value Length.
+    static final int MAX_KEY_LENGTH = 8 * 1024; // 8KB
+    private static final int MAX_SERIALIZATION_LENGTH = 1024 * 1024; // 1MB
+    private static final byte CURRENT_SERIALIZATION_VERSION = 0;
```

Comment:
I'm not sure this version makes sense. We are not introducing here the versioning scheme we have for compatibility, in which case, I'd assume that the compatibility of entries is at the table level: all entries must have the same version.  If my understanding is correct, then the version validation should be done at the table level.

Reply:
Every entry is serialized using EntrySerializer. It writes a header, followed by key then value. THe header contains the version
I didn't use VersionedSerializer because in most cases I only need to read the key, and not the value. VersionedSerializer requires me to read the whole thing every time.